server/api/main.py

```diff
@@ -277,15 +277,19 @@ async def _verify_log_collection_started_on_startup(
     db_session = await fastapi.concurrency.run_in_threadpool(create_session)
     log_collection_cycle_tracker = (
         server.api.utils.time_window_tracker.TimeWindowTracker(
-            key="log_collection_cycle",
+            key=server.api.utils.time_window_tracker.TimeWindowTrackerKeys.log_collection,
             max_window_size_seconds=min(
                 int(config.log_collector.api_downtime_grace_period),
                 int(config.runtime_resources_deletion_grace_period),
             ),
         )
     )
-    log_collection_cycle_tracker.initialize(db_session)
-    last_update_time = log_collection_cycle_tracker.get_window(db_session)
+    await fastapi.concurrency.run_in_threadpool(
```

Comment:
you might need to use `run_function_with_new_db_session` when running in different threads